tensorflow/core/ops/nn_ops.cc

```diff
@@ -1303,7 +1341,7 @@ output: The gradients for LRN.
 // --------------------------------------------------------------------------
 
 REGISTER_OP("MaxPool")
-    .Attr("T: {float, half} = DT_FLOAT")
+    .Attr("T: realnumbertype")
```

Comment:
The problem was with `MaxPool3dGrad`. We discussed why it is necessary to have two types for input and output, so compatibility tests can pass. I guess during rebase I dropped those changes by accident.